src/submission/merge-submission.ts

```diff
@@ -0,0 +1,105 @@
+import { DeepReadonly } from 'deep-freeze';
+import { Donor, Specimen, Sample } from '../clinical/clinical-entities';
+import { ActiveClinicalSubmission } from './submission-entities';
+import { FileType } from './submission-api';
+import _ from 'lodash';
+import { loggerFor } from '../logger';
+import { update } from './schema-api';
+const L = loggerFor(__filename);
+
+type ClinicalEnitityRecord = DeepReadonly<
+  Readonly<{
+    [key: string]: string;
+  }>
+>;
+
+export const mergeActiveSubmissionWithDonors = async (
```

Comment:
could call this function with a DonorMap instead of a donor array

Reply:
Do you see a need to refactor this way or is it a matter of preference?

I think, if activeSubmission merging is slow this could be a future optimization. would want to make a general Array->Map method to handle this type of situation.

Reply:
was just a suggestion, not critical to refactor.